gammapy/datasets/fermi.py

```diff
@@ -215,6 +217,158 @@ def fetch_fermi_diffuse_background_model(filename='gll_iem_v02.fit'):
 
     return filename
 
+class Fermi3FGLObject(object):
+    """
+    Class representing an object in the Fermi 3FGL catalog.
+    """
+
+    from astropy.units import Quantity
+    from ..spectrum import EnergyBounds
+
+    # Fermi catalog is lazily loaded on first access
+    # and cached at class level (not instance level)
+    fermi_cat = None
+
+    x_bins_edges = Quantity([30, 100, 300, 1000, 3000, 10000, 100000], 'MeV')
+
+    x_bins = Quantity(x_bins_edges, 'MeV')
+
+    x_cens = EnergyBounds(x_bins).log_centers
+
+    y_labels = ['Flux30_100', 'Flux100_300', 'Flux300_1000',
+                'Flux1000_3000', 'Flux3000_10000', 'Flux10000_100000']
+
+    def __init__(self, source_name):
+        fermi_cat = self.get_fermi_cat()
+        self.name_3FGL = source_name
+        self.catalog_index = np.where(self.fermi_cat[1].data['Source_Name'] == source_name)[0][0]
+        self.cat_row = fermi_cat[1].data[self.catalog_index]
+        self.ra = self.cat_row['RAJ2000']
+        self.dec = self.cat_row['DEJ2000']
+        self.glon = self.cat_row['GLON']
+        self.glat = self.cat_row['GLAT']
+        self.int_flux = self.cat_row['Flux_Density']
+        self.unc_int_flux = self.cat_row['Unc_Flux_Density']
+        self.spec_type = self.cat_row['SpectrumType']
+        self.pivot_en = self.cat_row['PIVOT_ENERGY']
+        self.spec_index = self.cat_row['Spectral_Index']
+        self.unc_spec_index = self.cat_row['Unc_Spectral_Index']
+        self.beta = self.cat_row['beta']
+        self.unc_beta = self.cat_row['unc_beta']
+        self.cutoff = self.cat_row['Cutoff']
+        self.unc_cutoff = self.cat_row['Unc_Cutoff']
+        self.exp_index = self.cat_row['Exp_Index']
+        self.unc_exp_index = self.cat_row['Unc_Exp_Index']
+        self.signif = self.cat_row['Signif_Avg']
+
+    @classmethod
+    def get_fermi_cat(cls):
```

Comment:
Make this a private method, i.e. add an underscore to the method name: `_get_fermi_cat`
Users should get the catalog via `fetch_fermi_catalog` ... this classmethod is an implementation detail they shouldn't see in the API docs.